src/main/java/org/openstreetmap/atlas/checks/validation/areas/WaterbodyAndIslandSizeCheck.java

```diff
@@ -0,0 +1,310 @@
+                }
+                else if (surfaceAreaKilometers > this.waterbodyMaximumArea)
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(1, object.getOsmIdentifier(),
+                                    surfaceAreaKilometers, this.waterbodyMaximumArea)));
+                }
+            }
+            // place=island or place=archipelago tags
+            if (IS_ISLAND.test(object))
+            {
+                if (surfaceAreaMeters < this.islandMinimumArea)
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(2,
+                            object.getOsmIdentifier(), surfaceAreaMeters, this.islandMinimumArea)));
+                }
+                else if (surfaceAreaKilometers > this.islandMaximumArea)
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(THREE, object.getOsmIdentifier(),
+                                    surfaceAreaKilometers, this.islandMaximumArea)));
+                }
+            }
+            // place=islet tag
+            if (IS_ISLET.test(object))
+            {
+                if (surfaceAreaMeters < this.isletMinimumArea)
+                {
+                    return Optional.of(this.createFlag(object, this.getLocalizedInstruction(FOUR,
+                            object.getOsmIdentifier(), surfaceAreaMeters, this.isletMinimumArea)));
+                }
+                else if (surfaceAreaKilometers > this.isletMaximumArea)
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(FIVE, object.getOsmIdentifier(),
+                                    surfaceAreaKilometers, this.isletMaximumArea)));
+                }
+            }
+        }
+        // There are two Relations to examine:
+        // 1. Islands included as members of a MultiPolygon (not otherwise tagged as place=island or
+        // place=islet)
+        // 2. Water bodies included as members of a Multipolygon (not otherwise tagged as
+        // natural=water)
+        // Each Relation will have n numbers of instructions, dependent on the amount of members
+        // that fail the size check
+        else if (object instanceof Relation)
+        {
+            final Relation relation = (Relation) object;
+            // Get Relation members that are an Area, have either an inner or outer member,
+            // and have yet to be examined
+            final Set<RelationMember> relationMembers = relation.members().stream()
+                    .filter(member -> member.getEntity() instanceof Area)
+                    .filter(member -> !this.isFlagged(member.getEntity().getIdentifier()))
+                    .filter(this::isInnerOrOuterMember).collect(Collectors.toSet());
```

Comment:
`filter` calls could be combined into one. There is no need to call `filter` method with different conditions if you can combine conditions into one. (Although compiler will probably do the similar optimization here as well.)